src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/SelectResultSet.java

```diff
@@ -555,8 +561,21 @@ private boolean hasTermAggregations(List<Aggregation> aggs) {
      */
     private Map<String, Object> addNumericAggregation(List<Aggregation> aggs, Map<String, Object> data) {
         for (Aggregation aggregation : aggs) {
-            NumericMetricsAggregation.SingleValue numericAggs = (NumericMetricsAggregation.SingleValue) aggregation;
-            data.put(numericAggs.getName(), numericAggs.value());
+            if (aggregation instanceof NumericMetricsAggregation.SingleValue){
+                NumericMetricsAggregation.SingleValue singleValueAggregation = (NumericMetricsAggregation.SingleValue) aggregation;
+                data.put(singleValueAggregation.getName(), !Double.isInfinite(singleValueAggregation.value()) ? singleValueAggregation.getValueAsString() : "null");
+            }
+            else if (aggregation instanceof Percentiles) {
+                Percentiles percentiles = (Percentiles) aggregation;
+
+                data.put(percentiles.getName(), StreamSupport
+                        .stream(percentiles.spliterator(), false)
```

Comment:
I'm not familiar with `Spliterator`. I'm curious what's the benefit we get from it? Anything special about `Percentile`?

And I notice `Percentile` doesn't override it and here is what I found from the Javadoc in `Iterable` interface. Any concern?

```
     * The default implementation should usually be overridden.  The
     * spliterator returned by the default implementation has poor splitting
     * capabilities, is unsized, and does not report any spliterator
     * characteristics. Implementing classes can nearly always provide a
     * better implementation.
     ...
    default Spliterator<T> spliterator() {
        return Spliterators.spliteratorUnknownSize(iterator(), 0);
    }
```

Reply:
You can't stream over ordinary Iterable, only foreach. The streamsupport enables you to do that

Reply:
> You can't stream over ordinary Iterable, only foreach. The streamsupport enables you to do that

I see. I was confused by the name `spliterator` and thought there are some parallelism benefits. Guava seems have  `Streams.stream(Iterable)` which looks much clearer to me. Unfortunately our version is too low.

Reply:
2nd parameter (false) disables parallelism :)